internal/repository/halls/hall.go

```diff
@@ -0,0 +1,117 @@
+package hall
+
+import (
+	"database/sql"
+	"fmt"
+
+	sq "github.com/Masterminds/squirrel"
+)
+
+type Repository struct {
+	DB *sql.DB
+}
+
+type Resource struct {
+	ID    int  `json:"ID"`
+	VIP   bool `json:"VIP"`
+	Seats int  `json:"seats"`
+}
+
+// Create new Hall in DB
+func (r *Repository) Create(hall Resource) (dbHall Resource, e error) {
+	var id int
+
+	query := sq.
+		Insert("halls").
+		Columns("vip", "seats").
+		Values(hall.VIP, hall.Seats).
+		Suffix("RETURNING \"id\"").
+		PlaceholderFormat(sq.Dollar).
+		RunWith(r.DB)
+
+	err := query.
+		QueryRow().
+		Scan(&id)
+	if err != nil {
+		return Resource{}, err
```

Comment:
👀

Reply:
/// You can retrieve id ...
query := sq.Insert("nodes").
    Columns("uuid", "type", "data").
    Values(node.Uuid, node.Type, node.Data).
    Suffix("RETURNING \"id\"").
    RunWith(m.db).
    PlaceholderFormat(sq.Dollar)

query.QueryRow().Scan(&node.id)

example in squirrel repo, used this one

Reply:
Yeah, this one is good, but why are you returning empty obj instead of nil? 
and have `func .... (*Resource, error)` in signature

Reply:
![image](https://user-images.githubusercontent.com/70835145/152045414-ab795c72-123b-40f1-a9ca-3e80a0a61a4f.png)

Reply:
![image](https://user-images.githubusercontent.com/34943627/152047864-9f6c0651-f217-4223-8338-17fed519e174.png)

Reply:
![image](https://user-images.githubusercontent.com/34943627/152048407-76a8c9d3-5c66-433f-b064-fcaf5d744ef2.png)

Reply:
![image](https://user-images.githubusercontent.com/34943627/152048433-8614de70-e66e-4149-9069-f7a6f9f3e16e.png)

Reply:
![image](https://user-images.githubusercontent.com/34943627/152048455-1b29abcd-7838-4434-b722-caafd4edf86e.png)